### PR TITLE
[SIE-151] Reorganize experiment fields

### PR DESCRIPTION
**Pull request by fernandowinfield**
### TLDR:
- Reorganized the fields of the Firestore **experiment** doc data model to be able to display more useful data in the Experiments page.
- Resolved pending TODOs in ExperimentsPage component.

*****

### In this PR:
- Updates to the experiment data model:
    - Removed fields:
        - admin
        - researcher
    - New fields:
        - creator
        - preSurvey
        - postSurvey
        - isOngoing

- Updates to Card component:
    - reduced most of the params of <Card/> into a new object param `experimentInfo`.
    - added new experiment info to be displayed when the ‘View details’ button is clicked (here’s where the researchers can access the link to our website that they can post in external platforms like SONA and mTurk).

- Updates to `getAllExperiments()` API:
    - Added the experiment doc ID to the response that’s returned so we can provide the link to researchers.

- Updates to AddExperiment component:
    - Added the new fields to the object that we post to Firestore.

- Updated to Experiments page:
    - Replaced the old auth with the new auth context.
    - Fixed logic to filter ongoing vs inactive experiments.

### CODE REVIEW

**Comment by fernandowinfield** [comment]
@theodore3131, tagging you here just in case so you are aware of this change I made to the experiment doc data model

File: src/stories/pages/ExperimentsPage/ExperimentsPage.js [hunk 1]
```diff
@@ -38,92 +28,87 @@ export const UserContext = React.createContext();
...
-        },
-        i,
-      ) =>
+  // Generate slider's slides for ongoing experiments (if any)
+  const ongoingExperimentsSlides = (ongoingExperiments &&
+    ongoingExperiments.length > 0) ?
+    ongoingExperiments.map((exp, i) => {
+      const experimentInfo = {
+        title: exp.title,
+        body: exp.description.length > MAX_DESCRIPTION_LENGTH ?
```
Comment (stenizwahyudi) [inline src/stories/pages/ExperimentsPage/ExperimentsPage.js:59]: I think let's just use 'body: exp.description' here. Trimming it doesn't seem to work well

File: src/stories/pages/ExperimentsPage/ExperimentsPage.js [hunk 2]
```diff
@@ -1,19 +1,12 @@
...
 import { AddExperiment } from '../AddExperiment/AddExperiment';
 // eslint-disable-next-line max-len
 import { HorizontalTitle } from '../../components/HorizontalTitle/HorizontalTitle';
 import { getAllExperiments } from '../../../firebase/api/experiments';
-// TODO #1.1(fernandowinfield): remove the import below.
-import UserServices from '../../../firebase/CRUDServices/userServices';
 import { Constrain } from '../../layouts/Constrain/Constrain';
+import { useAuth } from '../../../contexts/auth-provider';
 
 const MAX_DESCRIPTION_LENGTH = 100;
```
Comment (stenizwahyudi) [inline src/stories/pages/ExperimentsPage/ExperimentsPage.js:11]: Don't forget to delete this.

File: src/stories/pages/ExperimentsPage/ExperimentsPage.js [hunk 3]
```diff
@@ -38,92 +28,87 @@ export const UserContext = React.createContext();
...
+        postSurvey: exp.postSurvey,
+        experimentUrl:
+          `http://localhost:3000/${exp.experimentId}/<participantID>`,
+      };
+
+      return (
         <Card
           modifierClasses='card--active'
-          title={title}
-          body= {description.length > MAX_DESCRIPTION_LENGTH ?
```
Comment (stenizwahyudi) [inline src/stories/pages/ExperimentsPage/ExperimentsPage.js:85]: This one too.

File: src/stories/pages/ExperimentsPage/ExperimentsPage.js [hunk 4]
```diff
@@ -38,92 +28,87 @@ export const UserContext = React.createContext();
...
-        researchers={researchers.join(', ')}
-        key={i}
-      />);
+  // Generate slider's slides for inactive experiments (if any)
+  const inactiveExperimentsSlides = (inactiveExperiments &&
+    inactiveExperiments.length > 0) ?
+    inactiveExperiments.map((exp, i) => {
+      const experimentInfo = {
+        title: exp.title,
+        body: exp.description.length > MAX_DESCRIPTION_LENGTH ?
```
Comment (stenizwahyudi) [inline src/stories/pages/ExperimentsPage/ExperimentsPage.js:87]: Here too.

File: src/stories/components/Card/Card.js [hunk 5]
```diff
@@ -9,29 +9,52 @@ import { Button } from '../Button/Button';
...
+
+  // To show/hide experiment details
+  const toggleDetails = () => {
+    setShowDetails((prevState) => !prevState);
+  };
+
+  const cardDetails = showDetails ?
+    <React.Fragment>
+      <hr/>
+      <p>Consent form Qualtrics survey:</p>
```
Comment (stenizwahyudi) [inline src/stories/components/Card/Card.js:44]: This is simply a styling preferences, but I'd say it's better to use <h5>Consent form Qualtrics survey:</h5>.
Similarly on line 47, 50, and 53.

File: src/stories/pages/ExperimentsPage/ExperimentsPage.js [hunk 6]
```diff
@@ -1,19 +1,12 @@
...
-  TODOs
-    #1: Replace old user context with new auth context.
-    #2: Revise experiment information fields displayed in cards of slider.
-    #3: Divide fetched experiments into 'ongoing' and 'inactive'.
-*/
 import React, { useEffect, useState } from 'react';
 import { Slider } from '../../components/Slider/Slider';
 import { Card } from '../../components/Card/Card';
 import { AddExperiment } from '../AddExperiment/AddExperiment';
 // eslint-disable-next-line max-len
```
Comment (stenizwahyudi) [inline src/stories/pages/ExperimentsPage/ExperimentsPage.js:5]: Don't forget to delete this.

**Review by stenizwahyudi — CHANGES_REQUESTED** [review]
Please add <h5>Pre-Survey Questionnaire to line 103 (I believe) on AddExperiment.js. Similarly to Post-Survey Questionnaire. I thought I made the changes already, but not sure why it's not showing.

**Review by stenizwahyudi — APPROVED** [review]
